### ISSUE TITLE
waymo v1.0，AP result all 0, Number: (pd, 115125) VS. (gt, 0)

### ISSUE THREAD

**Issue by sinatayebati** [open-mmlab/OpenPCDet#1559]
I'm training on waymo v1.0 and the data infos are produced and available according to the instructions. However, in evaluation stage, it returns 0 for all AP. Also, it appears the ground truth box dict is empty [ (pd, 115125) VS. (gt, 0) ]. 

Has anyone faced the same issue and knows how to resolve this? Thanks in advance!

2020-12-23 10:36:51,746 INFO *************** EPOCH 30 EVALUATION *****************
eval: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1983/1983 [02:47<00:00, 11.87it/s, recall_0.3=(0, 54150) / 74830]
2020-12-23 10:39:38,759 INFO *************** Performance of EPOCH 30 *****************
2020-12-23 10:39:38,759 INFO Generate label finished(sec_per_example: 0.0842 second).
2020-12-23 10:39:38,759 INFO recall_roi_0.3: 0.000000
2020-12-23 10:39:38,759 INFO recall_rcnn_0.3: 0.723640
2020-12-23 10:39:38,759 INFO recall_roi_0.5: 0.000000
2020-12-23 10:39:38,760 INFO recall_rcnn_0.5: 0.640532
2020-12-23 10:39:38,760 INFO recall_roi_0.7: 0.000000
2020-12-23 10:39:38,760 INFO recall_rcnn_0.7: 0.400909
2020-12-23 10:39:38,760 INFO Average predicted number of objects(1983 samples): 58.056
Start the waymo evaluation...
Number: (pd, 115125) VS. (gt, 0)
Level 1: 0, Level2: 0)
2020-12-23 10:39:39.267625: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1159] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-12-23 10:39:39.267658: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1165]
2020-12-23 10:39:39.318131: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:39.318164: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:39.344263: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:39.460312: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:39.518959: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:39.518993: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:39.541163: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:39.655748: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:39.715529: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:39.715561: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:39.738099: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:39.852261: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:39.909590: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:39.909636: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:39.930640: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:40.057543: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:40.121250: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:40.121286: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:40.146768: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:40.264894: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:40.322628: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:40.322659: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:40.345940: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:40.463075: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:40.523954: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:40.523996: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:40.547976: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:40.661599: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:40.721689: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:40.721749: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:40.744914: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:40.856738: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:40.914816: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:40.914849: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:40.937291: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:41.055676: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:41.116400: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:41.116433: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:41.138756: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:41.256524: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:41.314279: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:41.314317: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:41.336739: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:41.460779: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:41.518792: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:41.518823: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:41.541227: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:41.656426: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:41.717149: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:41.717182: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:41.739818: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:41.853856: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:41.914684: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:41.914716: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:41.937205: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:42.052602: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:42.117211: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:42.117236: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:42.141252: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:42.262865: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:42.324895: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:157] Computing detection metrics for 115125 predicted boxes.
2020-12-23 10:39:42.324978: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:159] Parsing prediction [115125,7][115125]
2020-12-23 10:39:42.350499: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:168] Parsing ground truth [0,7][0]
2020-12-23 10:39:42.472671: I waymo_open_dataset/metrics/ops/detection_metrics_ops.cc:214] Done with computing detection metrics.
2020-12-23 10:39:42,531 INFO
OBJECT_TYPE_TYPE_VEHICLE_LEVEL_1/AP: 0.0000
OBJECT_TYPE_TYPE_VEHICLE_LEVEL_1/APH: 0.0000
OBJECT_TYPE_TYPE_VEHICLE_LEVEL_2/AP: 0.0000
OBJECT_TYPE_TYPE_VEHICLE_LEVEL_2/APH: 0.0000
OBJECT_TYPE_TYPE_PEDESTRIAN_LEVEL_1/AP: 0.0000
OBJECT_TYPE_TYPE_PEDESTRIAN_LEVEL_1/APH: 0.0000
OBJECT_TYPE_TYPE_PEDESTRIAN_LEVEL_2/AP: 0.0000
OBJECT_TYPE_TYPE_PEDESTRIAN_LEVEL_2/APH: 0.0000
OBJECT_TYPE_TYPE_SIGN_LEVEL_1/AP: 0.0000
OBJECT_TYPE_TYPE_SIGN_LEVEL_1/APH: 0.0000
OBJECT_TYPE_TYPE_SIGN_LEVEL_2/AP: 0.0000
OBJECT_TYPE_TYPE_SIGN_LEVEL_2/APH: 0.0000
OBJECT_TYPE_TYPE_CYCLIST_LEVEL_1/AP: 0.0000
OBJECT_TYPE_TYPE_CYCLIST_LEVEL_1/APH: 0.0000
OBJECT_TYPE_TYPE_CYCLIST_LEVEL_2/AP: 0.0000
OBJECT_TYPE_TYPE_CYCLIST_LEVEL_2/APH: 0.0000

2020-12-23 10:39:42,531 INFO Result is save to /home/zombie/zombie/deepleaning/lidar/OpenPCDet/output/waymo_models/second/default/eval/eval_with_train/epoch_30/val
2020-12-23 10:39:42,531 INFO Evaluation done.*
2020-12-23 10:39:42,538 INFO Epoch 30 has been evaluated
2020-12-23 10:40:12,565 INFO **********************End evaluation waymo_models/second(default)**********************o_models/second/default/ckpt `

**Comment by fx110127** [comment]
have you fixed this issue? I met the same question

**Comment by sinatayebati** [comment]
@fx110127 yes, the problem was using waymo v1.0. Start downloading waymo v1.2 instead, then produce data infos from scratch. Next go ahead with your training usual and you will get the correct numbers at the evaluation.
